mobile_rewards/src/server.rs

```diff
@@ -371,7 +397,92 @@ impl Server {
     }
 }
 
+const CONNECT_TIMEOUT: StdDuration = StdDuration::from_secs(5);
+const RPC_TIMEOUT: StdDuration = StdDuration::from_secs(5);
+const DEFAULT_URI: &str = "http://127.0.0.1:8080";
+
 async fn update_last_follower(pool: &Pool<Postgres>, last_height: u64) -> Result {
     metrics::gauge!("reward_server_last_reward_height", last_height as f64);
     Meta::update(pool, "last_follower_height", last_height.to_string()).await
 }
+
+fn new_client_from_env() -> Result<follower::Client<Channel>> {
+    let uri: Uri = env::var("FOLLOWER_URI")
+        .unwrap_or_else(|_| DEFAULT_URI.to_string())
+        .parse()?;
+    let channel = Endpoint::from(uri)
+        .connect_timeout(CONNECT_TIMEOUT)
+        .timeout(RPC_TIMEOUT)
+        .connect_lazy();
+    Ok(follower::Client::new(channel))
+}
+
+async fn txn_stream<T>(
+    client: &mut follower::Client<Channel>,
+    height: u64,
+    txn_hash: &[u8],
+    txn_types: &[T],
+) -> Result<Streaming<FollowerTxnStreamRespV1>>
+where
+    T: ToString,
+{
+    let req = FollowerTxnStreamReqV1 {
+        height,
+        txn_hash: txn_hash.to_vec(),
+        txn_types: txn_types.iter().map(|e| e.to_string()).collect(),
+    };
+    let res = client.txn_stream(req).await?.into_inner();
+    Ok(res)
+}
+
+async fn reward_period(client: &mut follower::Client<Channel>) -> Result<Range<u64>> {
+    let req = FollowerSubnetworkLastRewardHeightReqV1 {
+        token_type: BlockchainTokenTypeV1::from(ProtoTokenType::Mobile).into(),
+    };
+    let res = client
+        .subnetwork_last_reward_height(req)
+        .await?
+        .into_inner();
+
+    Ok(res.reward_height + 1..res.height)
+}
+
+pub fn get_time_range(last_reward_end_time: i64) -> (DateTime<Utc>, DateTime<Utc>) {
+    let after_utc = datetime_from_epoch(last_reward_end_time);
+    let now = Utc::now();
+    let stop_utc = now - Duration::minutes(DEFAULT_LOOKUP_DELAY);
+    let start_utc = after_utc.min(stop_utc);
+    (start_utc, stop_utc)
+}
+
+pub fn construct_txn(
+    keypair: &Keypair,
+    rewards: Vec<SubnetworkReward>,
+    period: Range<u64>,
+) -> Result<(BlockchainTxnSubnetworkRewardsV1, String)> {
+    let mut txn = BlockchainTxnSubnetworkRewardsV1 {
+        rewards: rewards,
+        token_type: BlockchainTokenTypeV1::from(ProtoTokenType::Mobile).into(),
+        start_epoch: period.start,
+        end_epoch: period.end,
+        reward_server_signature: vec![],
+    };
+    txn.reward_server_signature = sign_txn(&txn, keypair)?;
+    let hash = hash_txn_b64_url(&txn);
+    Ok((txn, hash))
+}
+
+use sha2::{Digest, Sha256};
```

Comment:
since this isn't scoped within a submodule or a function is there any conventional guideline around keeping all the use statements grouped together at the top of the file?

Reply:
I would not recommend doing this, I just through rustfmt would put it at the top of the file. 
If this sort of thing makes sense then I think that it probably makes more sense to split up into another module